app/src/main/java/com/poupa/vinylmusicplayer/misc/queue/StaticPlayingQueue.java

```diff
@@ -0,0 +1,461 @@
+
+        if (deletedPosition < position) {
+            this.currentPosition = position - 1;
+        } else if (deletedPosition == position) { //the current position was deleted
+            if (queue.size() > deletedPosition) {
+                return position;
+            } else {
+                return position - 1;
+            }
+        }
+
+        return -1;
+    }
+
+    // remove song at index position, numbering need to be redone for every song after this position (-1)
+    public int remove(int position) {
+        IndexedSong o = queue.remove(position);
+        backupQueue.remove(o.index);
+
+        for (int i = 0; i < queue.size(); i++) {
+            backupQueue.get(i).index = i;
+            if (queue.get(i).index >= o.index) {
+                queue.get(i).index = queue.get(i).index - 1;
+            }
+        }
+
+        songsIsStale = true;
+
+        return rePosition(position);
+    }
+
+    private int removeAllOccurrence(Song song) {
+        int hasPositionChanged = -1;
+
+        for (int i = queue.size() - 1; i >= 0; i--) {
+            if (queue.get(i).song.id == song.id) {
+                int temp = remove(i);
+                if (temp != -1) {
+                    hasPositionChanged = temp;
+                }
+            }
+        }
+
+        songsIsStale = true;
+
+        return hasPositionChanged;
+    }
+
+    public int removeSongs(@NonNull List<Song> songs) {
+        int hasPositionChanged = -1;
+
+        for (Song song : songs) {
+            int temp = removeAllOccurrence(song);
+            if (temp != -1) {
+                hasPositionChanged = temp;
+            }
+        }
+
+        return hasPositionChanged;
+    }
+
+    public void clear() {
+        songs.clear();
+        queue.clear();
+        backupQueue.clear();
+    }
+
+    private void revert() {
+        queue = new ArrayList<>(backupQueue);
+        songsIsStale = true;
+    }
+
+    /* -------------------- queue getter info -------------------- */
+
+    public boolean openQueue(@Nullable final ArrayList<Song> playingQueue, final int startPosition, final boolean startPlaying, int shuffleMode) {
+        if (playingQueue != null && !playingQueue.isEmpty() && startPosition >= 0 && startPosition < playingQueue.size()) {
+            clear();
+            addAll(playingQueue);
+
+            this.currentPosition = startPosition;
+
+            setShuffle(shuffleMode);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    public ArrayList<IndexedSong> getPlayingQueue() {
+        return queue;
+    }
+
+    public ArrayList<IndexedSong> getOriginalPlayingQueue() {
+        return backupQueue;
+    }
+
+    private void resetSongs() {
+        this.songs.clear();
+
+        for (IndexedSong song : queue) {
+            songs.add(song.song);
+        }
+        songsIsStale = false;
+    }
+
+    public ArrayList<Song> getPlayingQueueSongOnly() {
+        if (songsIsStale)
```

Comment:
Can you please leave some comment in the code (here or where the field is declared) explaining the intended use of `songsIsStale`? I dont get the idea

Reply:
seems like a waste to recalculate the output list every time this function is called, 
thus the songsIsStale is there to ensure we calculate it only if queue as been changed

Reply:
it is perhaps not that big of a deal